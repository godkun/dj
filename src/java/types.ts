import { JavaClazz } from './core/clazz'

export interface IType {
  tsType: string
  javaType: string
  generic?: Array<JavaClazz | IType>
}

export interface IJavaClazzField {
  name: string
  type: IType
  comment?: string
}

export type TServiceMethod = {
  [k in string]: IMethodMeta
}

export type IMethodArgs = {
  name: string
  type: string
}

export interface IMethodMeta {
  args: Array<IMethodArgs>
  ret: string
}
