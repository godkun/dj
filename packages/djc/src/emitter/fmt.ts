import p from 'prettier'

/**
 * format generate code
 * @param str
 */
export default function fmt(str: string) {
  return p.format(str, {
    trailingComma: 'es5',
    tabWidth: 2,
    semi: false,
    singleQuote: true,
    parser: 'typescript',
  })
}
