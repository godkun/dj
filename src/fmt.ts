import p from 'prettier'

export default function fmt(str: string) {
  return p.format(str, {
    trailingComma: 'es5',
    tabWidth: 2,
    semi: false,
    singleQuote: true,
    parser: 'typescript',
  })
}
