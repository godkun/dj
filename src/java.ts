import Deps from './java-deps'

export default class Base {
  deps: Deps = new Deps()

  cls2Path(cls: string) {
    return cls.replace(/\./g, '/')
  }

  buildComment(comment?: string) {
    return comment
      ? `
    /**
    * ${comment}
    */
   `
      : ''
  }

  get mott() {
    return `
      /**
       * auto generated by dubbo dj
       * ~~~ mechine coding ðŸ’— ~~~~
       */
    `
  }
}
