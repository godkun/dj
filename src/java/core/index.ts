/**
 * 通过构造Java Style code的DSL来自动生成 ts，java， go的dubbo框架的调用代码
 */

import Clazz from './clazz'
import Service from './service'
import {
  Integer,
  String,
  Boolean,
  List,
  Set,
  Collection,
  Map,
  boolean,
  int,
  short,
  Short,
  byte,
  Byte,
  long,
  Long,
  double,
  Double,
  float,
  Float,
  char,
  Char,
  chars,
  Character,
  Enumeration,
  Iterator,
  Currency,
  Dictionary,
  HashMap,
} from './java-type'

import Enum from './enum'

export default {
  Enum,
  Clazz,
  Service,

  Boolean,
  boolean,

  Integer,
  int,

  short,
  Short,

  byte,
  Byte,

  long,
  Long,

  double,
  Double,

  float,
  Float,

  String,
  char,
  Char,
  chars,
  Character,

  List,
  Set,
  Collection,
  Iterator,
  Enumeration,
  Map,
  HashMap,
  Dictionary,
  Currency,
}
