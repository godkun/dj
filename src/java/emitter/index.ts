import Clazz from './clazz'
import Enum from './enum'
import Service from './service'
import AbstractService from './service-abstract'

export { Enum, Clazz, Service, AbstractService }
