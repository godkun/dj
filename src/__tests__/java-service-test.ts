import fmt from '../fmt'
import java from '../'

const user = java
  .model('org.apache.dubbo.domain.User')
  .field('id', java.Integer)

const UserService = java
  .service('org.apache.dubbo.service.UserService')
  .group('A')
  .version('1.0.0')
  .method('sayHello')
  .arg('isExit', java.Boolean)
  .arg('user', user)
  .ret(java.Integer)
  .method('sayHi')
  .arg('i', java.Integer)
  .ret(java.String)

it('test java service', () => {
  expect(JSON.stringify(UserService, null, 2)).toMatchSnapshot()
  expect(fmt(UserService.serviceCode)).toMatchSnapshot()
})
