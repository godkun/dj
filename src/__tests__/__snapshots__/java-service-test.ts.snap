// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test java service 1`] = `
"{
  \\"deps\\": {
    \\"map\\": {}
  },
  \\"inf\\": \\"org.apache.dubbo.service.UserService\\",
  \\"_group\\": \\"\\",
  \\"_version\\": \\"1.0.0\\",
  \\"_methods\\": {
    \\"sayHello\\": {
      \\"args\\": [
        {
          \\"name\\": \\"isExit\\",
          \\"type\\": \\"boolean\\"
        },
        {
          \\"name\\": \\"user\\",
          \\"type\\": \\"User\\"
        }
      ],
      \\"ret\\": \\"number\\"
    },
    \\"sayHi\\": {
      \\"args\\": [
        {
          \\"name\\": \\"i\\",
          \\"type\\": \\"number\\"
        }
      ],
      \\"ret\\": \\"string\\"
    }
  },
  \\"_curMethodName\\": \\"sayHi\\",
  \\"_serviceName\\": \\"UserService\\"
}"
`;

exports[`test java service 2`] = `
"/**
 * auto generated by dubbo dj
 * ~~~ mechine coding ðŸ’— ~~~~
 */

import User from 'org/apache/dubbo/domain/User'

export default class UserService {
  dubboInterface = 'org.apache.dubbo.service.UserService'
  group = ''
  version = '1.0.0'
  sayHello(isExit: boolean, user: User): number {
    // TODO implements this this method
  }

  sayHi(isExit: boolean, user: User, i: number): string {
    // TODO implements this this method
  }
}
"
`;
